tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -423,8 +423,7 @@ void test_forever_timeout(void)
 	zassert_not_equal(k_sem_take(&tx_aborted, K_MSEC(1000)), 0,
 			  "TX_ABORTED timeout");
 	zassert_equal(k_sem_take(&tx_done, K_MSEC(100)), 0, "TX_DONE timeout");
-	zassert_not_equal(k_sem_take(&rx_rdy, K_MSEC(1000)), 0,
-			  "RX_RDY timeout");
+	zassert_equal(k_sem_take(&rx_rdy, K_MSEC(100)), 0, "RX_RDY timeout");
```

Comment:
when rx is started with `SYS_FOREVER_MS` timeout, it shall report RX_RDY event only if full buffer is recieved (100 bytes) so there should be no event after 5 bytes.